mlflow/projects/__init__.py

```diff
@@ -404,8 +404,10 @@ def _fetch_git_repo(uri, version, dst_dir):
         repo.heads.master.checkout()
 
 
-def _get_conda_env_name(conda_env_path):
+def _get_conda_env_name(conda_env_path, env_id=None):
     conda_env_contents = open(conda_env_path).read() if conda_env_path else ""
+    if env_id:
+        conda_env_contents += env_id
```

Comment:
What required this change? Should we document env_id? Should it be part of the conda env name instead?

Reply:
1. What required this change: To distinguish model environments with installed mlflow vs the ones without. If you run mlflow serve/predict ... --install-mlflow - mlflow ill be installed intot he environment. if you than run the same without --mlflow-install the mlflow ill still be there because the environment is cached. 

2. I guess we should, although it is an internal thing. I'll add a comment.

3. Could be, but I think it's better for it to be a part of the hash so that the mlflow environment names look uniform - makes it easier to grep and such.